engine/src/gradient.cpp

```diff
@@ -665,15 +665,16 @@ IO_stat MCGradientFillSerialize(MCGradientFill *p_gradient, MCObjectOutputStream
 		t_stat = p_stream . WriteU8(p_gradient -> repeat);
 	if (t_stat == IO_NORMAL)
 		t_stat = p_stream . WriteU8(p_gradient -> ramp_length);
-	
-	MCPoint *t_point;
-	t_point = &p_gradient -> origin;
-	for(int i = 0; i < 3 && t_stat == IO_NORMAL; ++i)
-	{
-		t_stat = p_stream . WriteS16(t_point[i] . x);
-		if (t_stat == IO_NORMAL)
-			t_stat = p_stream . WriteS16(t_point[i] . y);
-	}
+
+    auto t_write_point = [&p_stream, &t_stat](MCPoint& p) {
```

Comment:
@montegoulding Maybe we should find a good general solution for extending what can be written to an `MCObjectOutputStream` so that the code for serialising a particular type can live near the other functions that work on that type.  However, that is beyond the scope of what I was trying to achieve with this PR! 😉